### PR TITLE
Improving the documentation in forward-backward algorithm

### PR DESCRIPTION
**Pull request by igormq**
Better explanation on how the labels for CTC are generated (inclusive for non-aligned labels), because it is not straightforward as pointed in #2156, #1995, [StackOverflow](https://stackoverflow.com/questions/45699255/hello-world-ctc-connectionist-temporal-classification-model). The only documentation explaining the requirements is the line [328](https://github.com/Microsoft/CNTK/blob/10d7130a436158be9a1899b51d800bfb9f1334bc/Source/SequenceTrainingLib/gammacalculation.h#L328) in ```gammacalculation.h``` file.

Also, I have changed the input parameters to snake case, and some names in order to simplify the understanding.

### CODE REVIEW

File: bindings/python/cntk/ops/__init__.py [hunk 1]
```diff
@@ -261,28 +261,87 @@ def labels_to_graph(labels, name=''):
...
-        fb = C.forward_backward(graph, networkOut, 132)
+        labels_graph = cntk.labels_to_graph(labels_var)
+        network_out = model(input_var)
+        fb = C.forward_backward(labels_graph, network_out, 132)
+
+    Hint:
+        CNTK requires that both features and labels have the same time dimension.
+        This forward-backward code expects the duplication of these labels to
+        fulfill the requirements. The labels must be encoded as one-hot vectors,
+        where the positive index has values 1 or 2 (from now on referred to as
```
Comment (spandantiwari) [inline bindings/python/cntk/ops/__init__.py]: Did you mean "populated" instead of "positive"?

File: bindings/python/cntk/ops/__init__.py [hunk 2]
```diff
@@ -261,28 +261,87 @@ def labels_to_graph(labels, name=''):
...
+            # labels = [[2,0,0,...,0,0], [0,0,2,...,0,0], ..., [0,0,0,...,2,0]]
+
+            # Retrieve the input time dimension
+            time_dim = input.shape[-2]
+
+            expanded_labels = np.zeros((time_dim, labels.shape[-1]))
+            # We first copy the original one-hot labels
+            expanded_labels[:len(labels)] = labels
+
+            # Then, we replicate the last label as one-1-hot encoded
```
Comment (spandantiwari) [inline bindings/python/cntk/ops/__init__.py]: Spelling: "one-hot-1"

File: bindings/python/cntk/ops/__init__.py [hunk 3]
```diff
@@ -261,28 +261,87 @@ def labels_to_graph(labels, name=''):
...
+
+        Example:
+            from sklearn.preprocessing import LabelBinarizer
+
+            lb = LabelBinarizer(pos_label=2).fit(range(6))
+            labels = lb.transform([0, 2, 0, 1, 3, 4]) # blank is the label 5
+            # labels = [[2,0,0,...,0,0], [0,0,2,...,0,0], ..., [0,0,0,...,2,0]]
+
+            # Retrieve the input time dimension
+            time_dim = input.shape[-2]
```
Comment (spandantiwari) [inline bindings/python/cntk/ops/__init__.py]: Would it make sense to call this sequence_dim instead of time_dim?

File: bindings/python/cntk/ops/__init__.py [hunk 4]
```diff
@@ -261,28 +261,87 @@ def labels_to_graph(labels, name=''):
...
-        blankTokenId: id of the CTC blank label
-        delayConstraint: label output delay constraint introduced during training that allows to have shorter delay during inference. This is using the original time information to enforce that CTC tokens only get aligned within a time margin. Setting this parameter smaller will result in shorted delay between label output during decoding, yet may hurt accuracy. delayConstraint=-1 means no constraint
+        labels_graph: labels graph
+        network_out: network output
+        blank_token_id: id of the CTC blank label
+        delay_constraint: label output delay constraint introduced during
+            training that allows to have shorter delay during inference.
+            This is using the original time information to enforce that CTC
+            tokens only get aligned within a time margin. Setting this parameter
+            smaller will result in shorted delay between label output during
```
Comment (spandantiwari) [inline bindings/python/cntk/ops/__init__.py]: "shortened" instead of "shorted".

File: bindings/python/cntk/ops/__init__.py (hunk 1 above)
Comment (igormq) [reply bindings/python/cntk/ops/__init__.py]: Yes, you are right!

File: bindings/python/cntk/ops/__init__.py (hunk 3 above)
Comment (igormq) [reply bindings/python/cntk/ops/__init__.py]: Yes it would!

File: bindings/python/cntk/ops/__init__.py (hunk 4 above)
Comment (igormq) [reply bindings/python/cntk/ops/__init__.py]: Done!

**Comment by igormq** [comment]
Hi @spandantiwari , sorry for the delay, I was in college in the last couple of days.

I've made the modifications.

File: bindings/python/cntk/ops/__init__.py [hunk 5]
```diff
@@ -261,28 +261,87 @@ def labels_to_graph(labels, name=''):
...
+            # Retrieve the input time dimension
+            sequence_dim = input.shape[-2]
+
+            expanded_labels = np.zeros((sequence_dim, labels.shape[-1]))
+            # We first copy the original one-hot labels
+            expanded_labels[:len(labels)] = labels
+
+            # Then, we replicate the last label as one-hot-1 encoded
+            expanded_labels[len(labels):, labels[-1].argmax()] = 1
+            # expanded_labels = [[2,0,0,...,0,0], [0,0,2,...,0,0], ...,
```
Comment (spandantiwari) [inline bindings/python/cntk/ops/__init__.py]: Minor change. Can you please comment all these lines, including 309 and 310. That way it won't be an issue for anyone trying to run this code.

File: bindings/python/cntk/ops/__init__.py [hunk 6]
```diff
@@ -261,28 +261,87 @@ def labels_to_graph(labels, name=''):
...
+                                 [0,0,0,...,2,0], [0,0,0,...,1,0], ...,
+                                 [0,0,0,...,1,0]]
+
+            # We can define the model and the variables
+            input_var = sequence.input_variable((input.shape[-1]), name='input')
+            labels_var = sequence.input_variable((6), name='label')
+            # model = ...
```
Comment (spandantiwari) [inline bindings/python/cntk/ops/__init__.py]: A little elaboration here would be good. So maybe a full comment saying that the model should be defined here.
